### PR TITLE
type fixed and improved

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
Some typings were fixed and improved.
Exported enum so that users can use it
added a prettier config
changed the tsconfig
removed `.npmignore` as `files` in `package.json` will allow only `dist` files
used undici request instead of axios as it's faster
fixed a bug where `poweredBy: true` was removing the `poweredBy` footer

You can check commit wise to see all the changes one by one as all the commits has valid messages

### CODE REVIEW

**Comment by ItzDerock** [comment]
I'm not a huge fan of the larger tab size, can you revert it back to 2 spaces?

**Comment by imranbarbhuiya** [comment]
ig it's already in 2 spaces.

**Comment by ItzDerock** [comment]
you've set it to use the actual tab character instead of two spaces, and in your vsc you probably have some sort of setting that renders the tab character as two spaces.

File: tsconfig.json
Comment (ItzDerock) [inline tsconfig.json:3]: what versions of node support esnext target?

**Review by ItzDerock — CHANGES_REQUESTED** [review]

**Comment by imranbarbhuiya** [comment]
@ItzDerock  applied all the suggestions. Review once more. This pr is complete. And after u merge https://github.com/ItzDerock/discord-markdown-parser/pull/3 and release, I'll add support for that in a new pr

**Review by ItzDerock — APPROVED** [review]
lgtm!

**Review by ItzDerock — APPROVED** [review]
oops, forgot some files in my last review
everything else looks good!

**Comment by ItzDerock** [comment]
is there anything else you want to put in this pr, or is it ok to merge right now.

**Comment by imranbarbhuiya** [comment]
yeah u can merge it now. I'll create another pr with support for twemoji once u release new version of `discord-markdown-parser`.

**Pull request merged**
